ricecooker/classes/nodes.py

```diff
@@ -1,62 +1,102 @@
     files = [files] if isinstance(files, str) else files
     questions=[questions] if isinstance(questions, str) else questions
-    if files is not None and len(files) > 0:
+
+    # If there are any questions, return exercise
+    if questions is not None and len(questions) > 0:
+        return content_kinds.EXERCISE
+
+    # See if any files match a content kind
+    elif files is not None and len(files) > 0:
         for f in files:
             ext = f.rsplit('/', 1)[-1].split(".")[-1].lower()
             if ext in content_kinds.MAPPING:
                 return content_kinds.MAPPING[ext]
         raise InvalidFormatException("Invalid file type: Allowed formats are {0}".format([key for key, value in content_kinds.MAPPING.items()]))
-    elif questions is not None and len(questions) > 0:
-        return content_kinds.EXERCISE
+
+    # If there are no files/questions, return topic
     else:
         return content_kinds.TOPIC
 
-""" TreeModel: model to handle structure of channel """
-class TreeModel:
+
+class Node:
+    """ Node: model to represent all nodes in the tree """
     def __init__(self):
         self.children = []
 
-    """ to_dict: formats data to what CC expects
-        @return dict of model's data
-    """
+    def __str__(self):
+        pass
+
     def to_dict(self):
+        """ to_dict: puts data in format CC expects
+            Args: None
+            Returns: dict of channel data
+        """
         pass
 
-    """ add_child: adds children to root node
-        @param node (node to add to children)
-    """
     def add_child(self, node):
+        """ add_child: Adds child node to node
+            Args: node to add as child
+            Returns: None
+        """
         self.children += [node]
 
     def count(self):
+        """ count: get number of nodes in tree
+            Args: None
+            Returns: int
+        """
         total = len(self.children)
         for child in self.children:
             total += child.count()
         return total
 
     def print_tree(self, indent=1):
+        """ print_tree: prints out structure of tree
+            Args: indent (int): What level of indentation to start printing at
+            Returns: None
+        """
         print("{indent}{data}".format(indent="   " * indent, data=str(self)))
         for child in self.children:
             child.print_tree(indent + 1)
 
-    def __str__(self):
-        pass
+    def test_tree(self):
+        """ test_tree: validate all nodes in this tree
+            Args: None
+            Returns: boolean indicating if tree is valid
+        """
+        self.validate()
+        for child in self.children:
+            assert child.test_tree()
+        return True
+
+    def validate(self):
+        """ validate: Makes sure node is valid
+            Args: None
+            Returns: boolean indicating if node is valid
+        """
+        assert self.id is not None
+        assert isinstance(self.title, str)
```

Comment:
Please add the human-readable error messages for these. These assert statements will otherwise return something unintelligible.